src/qt/dcclient.cpp

```diff
@@ -88,10 +114,12 @@ wxClientDCImpl::~wxClientDCImpl()
     {
         m_qtPainter->end();
         m_ok = false;
-        QPicture *pict = m_window->QtGetPicture();
+
 
         if ( m_window != NULL )
         {
+            QtPictureSetter(m_window, m_pic);
```

Comment:
Sorry, but this can't work AFAICS as it creates and destroys the temporary object immediately, thus setting and unsetting the picture. The fix is trivial:
```suggestion
            QtPictureSetter pictureSetter(m_window, m_pic);
```
but what worries me is that this has somehow went unnoticed in testing, when it really shouldn't work at all.